### PR TITLE
DAOS-9161 Test: Update code to detect VMD capacity to be used by DAOS.

### PR DESCRIPTION
**Pull request by ravalsam**
- Updating the get_storage_capacity() to work when system has VMD.
  VMD device is not 1:1 mapped with NVMe and there are multiple NVMe device
  connected behind the VMD address, so need to calculate the total pool
  capacity available for testing purpose.
- Updated launch.py to run reset command as sometime system does not reset
  the device at the end of the test and that will cause the problem to detect
  NVMe drive behind VMD for testing.
- Remove --enable-vmd option from SPDK script as control plane do not required
  anymore.

Signed-off-by: Samir Raval <samir.raval@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/server_utils_base.py [hunk 1]
```diff
@@ -516,11 +515,23 @@ def get_storage_capacity(self, engine_params):
...
+            if "True" in os.environ.get("DAOS_ENABLE_VMD"):
+                # Get the NVMe storage behind the VMD address
+                for vmd_dev in bdev_list:
+                    device = (vmd_dev.split(":")[1] +
+                              vmd_dev.split(":")[2].split(".")[0]+
+                              vmd_dev.split(":")[2].split(".")[1].zfill(2))
+                    storage_capacity["nvme"].append(0)
+                    for key in device_capacity["nvme"]:
+                        if device in key:
+                            storage_capacity["nvme"][-1] = (storage_capacity["nvme"][-1]+
```
Comment (tanabarr) [inline src/tests/ftest/util/server_utils_base.py:527]: could you not use "+=" operator to simplify this line

**Review by tanabarr — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

**Comment by ravalsam** [comment]
The single test failure is an existing test failure because of DAOS-9670

**Review by phender — COMMENTED** [review]
@ravalsam we're testing VMD in one of the CI HW Medium cluster converted to use VMD in https://daosio.atlassian.net/browse/DAOS-9562.  It seems like it would be best to consolidate efforts in that ticket and related PR.

**Comment by ravalsam** [comment]
> @ravalsam we're testing VMD in one of the CI HW Medium cluster converted to use VMD in https://daosio.atlassian.net/browse/DAOS-9562. It seems like it would be best to consolidate efforts in that ticket and related PR.

Yes, sure for going forward will do that.
For this PR the defect was reported while ago DAOS-9161 so would suggest to land this and I have tested it on Australis cluster.

**Comment by ravalsam** [comment]
Closing this one as it will be taken care in another PR https://github.com/daos-stack/daos/pull/7868.

**Pull request closed without merging**